### PR TITLE
Improve rpmdb consistency check

### PR DESCRIPTION
**Pull request by fcrozat**
Check rpmdb is fine

Restrict zypper to local repository, it catches system db inconsistencies.

### CODE REVIEW

**Comment by sysrich** [comment]
LGTM and doesn't cause insane chaos with my Aeon images like the current code :)

**Comment by laenion** [comment]
Thanks, that's definitely better!

I'm still no entirely sure we want to have the second check though: You only get into a state with db inconsistencies if you explicitly confirm it during package installation. Two of my two production systems have a package installed where a dependency explicitly had to be broken (because the commercial applications were packaged for Red Hat based systems and the dependencies have a different name there).
Because you can only get into such a state when you explicitly confirm it beforehand I don't think this check is necessary, but on the contrary will roll back systems explicitly confirmed as good by the admin. Another case is https://bugzilla.suse.com/show_bug.cgi?id=1224773#c56, where an (so far) unconfirmed repository key caused a rollback.

What do you think?

File: plugins/rpmdb-consistency.sh [hunk 1]
```diff
@@ -1,7 +1,11 @@
 #!/bin/bash
 
 run_checks() {
-	zypper --no-refresh --quiet verify --dry-run
+	# check rpm DB itself
+	rpm -D "%_rpmlock_path /run/rpmdb"  --verifydb
```
Comment (Vogtinator) [inline plugins/rpmdb-consistency.sh]:
```suggestion
	rpm -D "%_rpmlock_path /run/rpmdb" --verifydb
```

and IMO this needs a comment why it sets `%_rpmlock_path`

File: plugins/rpmdb-consistency.sh [hunk 2]
```diff
@@ -1,7 +1,11 @@
...
+	test $? -ne 0 && exit 1
```
Comment (Vogtinator) [inline plugins/rpmdb-consistency.sh]: Please just `|| exit 1` if you touch this anyway

**Comment by fcrozat** [comment]
> Thanks, that's definitely better!
> 
> I'm still no entirely sure we want to have the second check though: You only get into a state with db inconsistencies if you explicitly confirm it during package installation. Two of my two production systems have a package installed where a dependency explicitly had to be broken (because the commercial applications were packaged for Red Hat based systems and the dependencies have a different name there). Because you can only get into such a state when you explicitly confirm it beforehand I don't think this check is necessary, but on the contrary will roll back systems explicitly confirmed as good by the admin. Another case is https://bugzilla.suse.com/show_bug.cgi?id=1224773#c56, where an (so far) unconfirmed repository key caused a rollback.
> 
> What do you think?

I'm still convinced the "having RPM db in inconsistent state" case is a corner case, but since it is not yet fully working, I can :
- move the zypper test to a separate plugin
- have this plugin in a separate subpackage, not shipped by default (at least for now) for MicroOS and Aeon

WDYT ?

**Comment by Vogtinator** [comment]
I agree that it's a corner case (and an effectively "unsupported" one at that), but I also think that it will catch more false positives than true positives. I don't see a way to end up with broken deps which weren't caused by either an interrupted transaction (which shouldn't happen on MicroOS) or by the user (semi-)intentionally.

Subpackage sounds like a good compromise.

**Comment by Vogtinator** [comment]
I'd like to get this in, can you do the split or comment out the zypper verify call for now?

**Comment by Vogtinator** [comment]
Unfortunately the current selinux-policy blocks this: https://bugzilla.opensuse.org/show_bug.cgi?id=1231127
